### ISSUE TITLE
Test Explorer UI Test Adapter for Visual Studio Code

### ISSUE THREAD

**Issue by thoni56** [cgreen-devs/cgreen#179]

**Comment by thoni56** [comment]
I've started work on one based of the [Ceedling](https://github.com/numaru/vscode-ceedling-test-adapter) one (https://github.com/numaru/vscode-ceedling-test-adapter/commit/8965506ddeb14b66ba1858c6d56fd45879521a2dis).

It is at [cgreen-devs/vscode-cgreen-test-adapter](https://github.com/cgreen-devs/vscode-cgreen-test-adapter).

**Comment by joaopapereira** [comment]
@thoni56 did you ever start this? I was looking at this and it think it is going to be a little bit hard for us, because Ceedling have the ability to group tests files and you can retrieve that information, while we do not.
I believe that in order to create this adapter we might need to do some changes on the runner in order to get the tests from the libraries or something like this

**Comment by thoni56** [comment]
Yes, I started it, but found what you also have seen, that copying the approach from the Ceedling adapter will not work. So I restarted from scratch, to not be influenced by that. But I was also halted by other priorities, projects, holidays and influenzas...

Any ideas and prototypes are welcome!

**Comment by joaopapereira** [comment]
There are 2 things I am not sure about:
1. How should we get the tests that need to be running
2. How can we find the libraries that have the compiled tests

These 2 questions are tough because we do not control the compilation. Can we have any of the following assumptions:
1. all test code is inside `test` or `tests` folder?
1. compiled shared libraries are in `build/tests/*.so` or `tests/*.so` or `test*.so`

**Comment by thoni56** [comment]
I haven't done anything about this since my initial idea. But I was thinking that we could assume that `cgreen-runner` was used, that thus inspect all *.so/.dll for tests. This would be the closest to the "truth" since those could be run and output matched to the names found.

We could then reuse the discovery part of the `cgreen-runner`, and use XML-output for results.

But that was just some ideas I got when realizing what you also did.
